Assets/MixedRealityToolkit.SDK/Experimental/Features/Utilities/TapToPlace/TapToPlace.cs

```diff
@@ -0,0 +1,467 @@
+            else
+            {
+                GoalPosition = SolverHandler.TransformTarget.position + (SolverHandler.TransformTarget.forward * DefaultPlacementDistance);
+            }
+        }
+
+        protected virtual void SetRotation()
+        {
+            Vector3 direction = currentRay.Direction;
+            Vector3 surfaceNormal = currentHit.normal;
+
+            if (KeepOrientationVertical)
+            {
+                direction.y = 0;
+                surfaceNormal.y = 0;
+            }
+
+            // if the object is on a surface then change the rotation according to the normal of the hit point
+            if (DidHit && rotateAccordingToSurface)
+            {
+                GoalRotation = Quaternion.LookRotation(-surfaceNormal, Vector3.up);
+            }
+            else 
+            {
+                // If (DidHit and !rotateAccordingToSurface) or if there is no raycast hit, 
+                GoalRotation = Quaternion.LookRotation(direction, Vector3.up);
+            }
+        }
+
+        private void ToggleSpatialMeshVisibility(bool spatialMeshVisibility)
+        {
+            // If the spatial mesh is already visible on start, then we do not need to toggle the spatial mesh 
+            // if the object is in the placing state
+            if (InitialSpatialMeshDisplayOption == SpatialAwarenessMeshDisplayOptions.None || InitialSpatialMeshDisplayOption == SpatialAwarenessMeshDisplayOptions.Occlusion)
+            {
+                IReadOnlyList<IMixedRealitySpatialAwarenessMeshObserver> observers = spatialAwarenessSystemAcecess.GetDataProviders<IMixedRealitySpatialAwarenessMeshObserver>();
+
+                foreach (IMixedRealitySpatialAwarenessMeshObserver observer in observers)
+                {
+                    // If the user wants the spatial mesh visible while in the placing state
+                    if (spatialMeshVisibility)
+                    {
+                        observer.DisplayOption = SpatialAwarenessMeshDisplayOptions.Visible;
+                    }
+                    else
+                    {
+                        observer.DisplayOption = InitialSpatialMeshDisplayOption;
+                    }
+                }
+            }
+        }
+
+        #region IMixedRealityPointerHandler
+
+        /// <inheritdoc/>
+        public void OnPointerDown(MixedRealityPointerEventData eventData) { }
+
+        /// <inheritdoc/>
+        public void OnPointerDragged(MixedRealityPointerEventData eventData) { }
+
+        /// <inheritdoc/>
+        public void OnPointerUp(MixedRealityPointerEventData eventData) { }
+
+        /// <inheritdoc/>
+        public void OnPointerClicked(MixedRealityPointerEventData eventData)
+        {
+
+            // When a click is called in the same second then it is a mistake and no action needs to be taken
+            Debug.Log("Time Difference: " + (Time.time - previousFrameNumber));
+            if ((Time.time - previousFrameNumber) < 1.0f)
```

Comment:
if you are trying to use delta time effectively, use 
https://docs.unity3d.com/ScriptReference/Time-realtimeSinceStartup.html

Reply:
I do not think realtimeSinceStartup will be helpful in this case.  I am trying to catch an extra OnPointerClicked call so the placing state is not switched back immediately.  The issue is explained [here](https://github.com/microsoft/MixedRealityToolkit-Unity/pull/7188#discussion_r376031285)